metis-common/src/main/java/eu/europeana/metis/zoho/ZohoAccessClient.java

```diff
@@ -53,27 +47,31 @@ public ZohoAccessClient() {
    * tokens from the persistence storage.
    * </p>
    *
-   * @param grantToken the grant token key to be used
+   * @param grantToken the grant token key to be used. Can be null
+   * @param zcrmConfigurations the key value map of zoho configuration
    * @throws ZohoOAuthException if initialization of {@link ZCRMRestClient} failed.
    */
-  public ZohoAccessClient(String grantToken) throws ZohoOAuthException {
+  public ZohoAccessClient(@Nullable String grantToken, HashMap<String, String> zcrmConfigurations)
+      throws ZohoOAuthException {
     try {
-      ZCRMRestClient.initialize();
+      ZCRMRestClient.initialize(zcrmConfigurations);
     } catch (Exception ex) {
       LOGGER.error("ZCRMRestClient failure!");
       throw new ZohoOAuthException(ex);
     }
-    ZohoOAuthClient cli = ZohoOAuthClient.getInstance();
-    try {
-      cli.generateAccessToken(grantToken);
-    } catch (ZohoOAuthException e) {
-      LOGGER.warn(
-          "THIS EXCEPTION IS PROBABLY NORMAL. THAT CAN HAPPEN IF THE GRANT TOKEN HAS BEEN SUCCESSFULLY USED ONCE ALREADY!"
-              + "Exception when generating access token. Grant tokens can be used only once, "
-              + "so when the access and refresh tokens are generated, the grant token becomes obsolete on subsequent deployments",
-          e);
+    //Try the grand token if available
+    if (StringUtils.isNotBlank(grantToken)) {
```

Comment:
Also here: can use `String.isBlank()`. I think there are other occurrences in this class. Or are you allowing for it to be null as well?

Reply:
Same as the other remark above